### PR TITLE
Upgrade to Django 3

### PR DESCRIPTION
**Pull request by oktaal**
It's also straightforward to use 2 instead. They only difference with Django 3 is that the plugin **django-revproxy** breaks. I've sent a PR to them:

https://github.com/TracyWebTech/django-revproxy/pull/102

As Django 3 is scheduled to be released this December I think it would be good to use 3 for new developments.

### CODE REVIEW

**Comment by jgonggrijp** [comment]
So, is the plan to wait with merging this until December or until https://github.com/TracyWebTech/django-revproxy/pull/102 is merged, whichever happens last?

File: {{cookiecutter.slug}}/backend/{{cookiecutter.slug}}/index.py [hunk 1]
```diff
@@ -9,6 +9,7 @@ def index(request):
     filepath = request.path[1:] or "index.html"
     static_path = finders.find(filepath)
     if not static_path:
-        raise Http404('Static file not found at: ' + filepath + '. Has the frontend been compiled?')
+        # a path which should be handled by the SPA
+        static_path = finders.find("index.html")
```
Comment (jgonggrijp) [inline {{cookiecutter.slug}}/backend/{{cookiecutter.slug}}/index.py]: Am I correctly interpreting this as a workaround for not being able (yet) to configure the static prefix in the frontend?

**Comment by oktaal** [comment]
I've reworked it so things should be passed through properly now. Even did away with npx. Currently it references the git-repository (master) of django-revproxy as they have merged the PR, but it isn't on PyPI yet. I think it's reasonably improbable that a project which will be started now will have to be completely finished before the release date of Django 3. This way any new project will be a bit more future-proof.

**Comment by oktaal** [comment]
Oh come on, Travis doesn't install the Django pre-release specified in the requirements.in for some reason. I give up for now, _it works on my machine™_. I'll look into it whenever a new project needs to be started.

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -18,7 +24,7 @@
 )
 PSQL_COMMAND = '{{cookiecutter.psql_command}}'
 VIRTUALENV = '{{cookiecutter.virtualenv}}'
-VIRTUALENV_COMMAND = '{{cookiecutter.virtualenv_command}}'
+VIRTUALENV_COMMAND = '{{cookiecutter.virtualenv_command}}'.replace('%PYTHON%', python_path())
```
Comment (jgonggrijp) [inline hooks/post_gen_project.py:27]: Clever. Maybe add `/opt/local/bin/python3` for MacPorts users. By the way, what happens if I override the string so it doesn't contain `%PYTHON%` anymore?

File: {{cookiecutter.slug}}/backend/{{cookiecutter.slug}}/index.py [hunk 3]
```diff
@@ -6,9 +6,4 @@
 @ensure_csrf_cookie
 def index(request):
     """ Thin wrapper for the static index.html that adds the CSRF cookie."""
-    filepath = request.path[1:] or "index.html"
-    static_path = finders.find(filepath)
-    if not static_path:
-        raise Http404('Static file not found at: ' + filepath + '. Has the frontend been compiled?')
-    with open(static_path, 'rb') as file:
-        return HttpResponse(file, content_type=mimetypes.guess_type(filepath)[0])
+    return HttpResponse(content=open(finders.find('index.html')))
```
Comment (jgonggrijp) [inline {{cookiecutter.slug}}/backend/{{cookiecutter.slug}}/index.py:9]: Hooray!

File: {{cookiecutter.slug}}/bootstrap.py [hunk 4]
```diff
@@ -184,6 +183,7 @@ def modify_angular_json():
         data = json.load(file)
     try:
         project = '{{cookiecutter.slug}}'.replace('_', '-')
+        data['projects'][project]['architect']['build']['options']['deployUrl'] = '/static/'
```
Comment (jgonggrijp) [inline {{cookiecutter.slug}}/bootstrap.py:186]: I was actually going to point you to a Stack Overflow answer that mentioned this solution. But you already found it yourself. 👍

**Review by jgonggrijp — APPROVED** [review]
I'm happy with these changes. As for the Travis build, that might fail because of the git+git url. Have you tried https+git (or git+https, not sure about the order)? This has caused issues with bower, npm and yarn in our puppet deployments, too.

File: hooks/post_gen_project.py (hunk 2 above)
Comment (oktaal) [reply hooks/post_gen_project.py:27]: Nothing! It will just use whatever command you specified. I've added the path for MacPorts!

**Pull request merged**